hail/python/hail/matrixtable.py

```diff
@@ -224,16 +227,133 @@ def partition_hint(self, n: int) -> 'GroupedMatrixTable':
         return self
 
     @typecheck_method(named_exprs=expr_any)
-    def aggregate(self, **named_exprs) -> 'MatrixTable':
-        """Aggregate by group, used after :meth:`.MatrixTable.group_rows_by` or :meth:`.MatrixTable.group_cols_by`.
+    def aggregate_cols(self, **named_exprs) -> 'GroupedMatrixTable':
+        """Aggregate cols by group.
+
+        Examples
+        --------
+        Aggregate to a matrix with cohort as column keys, computing the mean height
+        per cohort as a new column field:
+
+        >>> dataset_result = (dataset.group_cols_by(dataset.cohort)
+        ...                          .aggregate_cols(mean_height = agg.stats(dataset.pheno.height).mean)
+        ...                          .result())
+
+        Notes
+        -----
+        The aggregation scope includes all column fields and global fields.
+
+        See Also
+        --------
+        :meth:`.result`
+
+        Parameters
+        ----------
+        named_exprs : varargs of :class:`.Expression`
+            Aggregation expressions.
+
+        Returns
+        -------
+        :class:`.GroupedMatrixTable`
+        """
+
+        if self._row_keys:
+            raise NotImplementedError("GroupedMatrixTable is already grouped by rows. Cannot aggregate over cols.")
+
+        assert self._col_keys is not None
+
+        existing_fields = list(self._parent.globals.dtype)
+        existing_fields.extend(self._col_keys.keys())
+        existing_fields.extend(list(self._parent.row.dtype))
+        if self._entry_fields is not None:
+            existing_fields.extend(list(self._entry_fields.keys()))
+
+        new_fields = self._col_fields if self._col_fields is not None else {}
```

Comment:
I don't totally understand what's going on here - does this let you call `aggregate_cols` multiple times? If so, I think I'd prefer to just check that _col_fields is None and throw a nice error if it's not.

Reply:
Yes, this lets you call aggregate_cols multiple times. I was thinking of this operation as analogous to `annotate_cols`.

Reply:
I'm OK with this design, but should we then change the interface for Table?

Reply:
I didn't realize GroupedTable existed. I don't like how aggregate means an alias for `aggregateEntries.result` in the MatrixTable case but not for Table. What do you think of `aggregate_rows`, `result`, and `aggregate` as an alias for `aggregate_rows.result`? But none of the current table operations are `*_rows`...

Reply:
hmmm. I feel it'll be a bit symmetric either way. Let's leave it for now; this is a back-compatible change.

Reply:
you've said the important points for future discussions